src/backends/opentsdb.c

```diff
@@ -0,0 +1,117 @@
+#include <stddef.h>
+#include <string.h>
+#include "brubeck.h"
+
+static inline int is_connected(struct brubeck_opentsdb *self)
+{
+        return (self->out_sock >= 0);
+}
+
+static int opentsdb_connect(void *backend)
+{
+        struct brubeck_opentsdb *self = (struct brubeck_opentsdb *)backend;
+
+        if (is_connected(self))
+                return 0;
+
+        self->out_sock = socket(PF_INET, SOCK_STREAM, IPPROTO_TCP);
+
+        if (self->out_sock >= 0) {
+                int rc = connect(self->out_sock,
+                                 (struct sockaddr *)&self->out_sockaddr,
+                                 sizeof(self->out_sockaddr));
+
+                if (rc == 0) {
+                        log_splunk("backend=opentsdb event=connected");
+                        sock_enlarge_out(self->out_sock);
+                        return 0;
+                }
+
+                close(self->out_sock);
+                self->out_sock = -1;
+        }
+
+        log_splunk_errno("backend=opentsdb event=failed_to_connect");
```

Comment:
Strictly speaking the close() system call may overwrite errno, so you should probably log this just after connect() failed before calling close() and as an "else" block when socket() fails. That has the added benefit of indicating exactly which system call failed.